lib/plugins/aws/invokeLocal/resolvers/index.js

```diff
@@ -0,0 +1,160 @@
+    }
+
+    if(cache[action] && cache[action][resourceName]) {
+        switch(action) {
+            case 'Ref':
+            case 'Fn::ImportValue':
+                environment[name] = cache[action][resourceName];
+                return;
+            case 'Fn::GetAtt':
+                const val = cache[action][resourceName][value[action][1]];
+                if(val) {
+                    environment[name] = val;
+                    return;
+                }
+                break;
+        }
+    }
+
+    if(resourceName) {
+        let resource = serverless.service.resources.Resources[resourceName];
+        if(!resource && resourceName.endsWith('LambdaFunction')) {
+            let lambdaName = resourceName.substr(0, resourceName.length - 'LambdaFunction'.length);
+            let func = serverless.service.functions[lambdaName];
+            if(!func) {
+                const lowerLambdaName = lambdaName.substr(0, 1).toLowerCase() + lambdaName.substr(1, lambdaName.length - 1);
+                func = serverless.service.functions[lowerLambdaName];
+                lambdaName = lowerLambdaName;
+            }
+            if(func) {
+                resource = {
+                    Type: "AWS::Lambda::Function",
+                    Properties: {
+                        FunctionName: `${serverless.service.service}-${serverless.service.provider.stage}-${lambdaName}`
+                    }
+                };
+            }
+        }
+        if(resource) {
+            const handler = handlers[resource.Type];
+            if(handler) {
+                environment[name] = await handler(action, value[action], resourceName, resource, getCFResources, serverless, serviceConfig);
+            } else {
+                return;
+            }
+        }
+    } else if(importName) {
+        if(!cfExports) {
+            const cf = new aws.CloudFormation(serviceConfig);
+            cfExports = [];
+            let token = undefined;
+            do {
+                const result = await cf.listExports({ NextToken: token}).promise();
+                if(result.Exports) {
+                    cfExports.push(...result.Exports);
+                }
+                token = result.NextToken;
+            } while(token);
+        }
+
+        const exp = cfExports.find(x => x.Name == importName);
+        if(exp) {
+            environment[name] = exp.Value;
+        } else {
+            return;
+        }
+    }
+
+    if(!cache[action]) {
+        cache[action] = {};
+    }
+    switch(action) {
+        case 'Ref':
+        case 'Fn::ImportValue':
+            cache[action][resourceName] = environment[name];
+            return;
+        case 'Fn::GetAtt':
+            if(!cache[action][resourceName]) {
+                cache[action][resourceName] = {};
+            }
+            cache[action][resourceName][value[action][1]] = environment[name];
+            break;
+    }
+
+    fs.writeFileSync(cacheFilePath, JSON.stringify(cache));
```

Comment:
We should avoid relying any _sync_ counterparts of _async_ operations (I know there's tons of them across framework, but it became an issue for its performance, and we slowly revert from them)

Reply:
Going async on this could create write conflicts as multiple resources could try to save at the same time, which could be problematic.  I think above here, there could be an optimization to automatically setup all import references, as that is at an account level, but not at the resource level.  Saving the last cache would be important, not sure if we can get away from the writeFileSync here without repercussions.

Reply:
There are writeAtomic util we use, that makes handling of such race conditions safe

Reply:
Caching between invokes has been removed.  I'm not seeing a big enough performance increase to justify the other issues it creates